hi-ml/src/health_ml/model_trainer.py

```diff
@@ -255,6 +260,19 @@ def model_train(checkpoint_path: Optional[Path],
     assert trainer.logger is not None
     trainer.logger.finalize('success')
 
+    if container.run_extra_val_epoch:
+        if checkpoint_handler:
+            checkpoint_handler.additional_training_done()
+            checkpoint_path_for_inference = checkpoint_handler.get_checkpoint_to_test()
+            container.load_model_checkpoint(checkpoint_path_for_inference)
+            lightning_model = container.model
+
+        with logging_section("Additional validation epoch"):
+            assert hasattr(lightning_model, "run_extra_val_epoch"), "Model does not have run_extra_val_epoch flag."
+            "This is required for running an additional validation epoch to save plots."
```

Comment:
We want it to be set to False ([run_extra_val_epoch](https://github.com/microsoft/hi-ml/blob/ce1535b5958c10a1c883411578247c934befce9e/hi-ml-histopathology/src/histopathology/models/deepmil.py#L101)) during trainer.fit (that does validation as well) and only switched it on before trainer.validate so that the outputs handler gets that it's time to produce plots for validation as well 
[run_extra_val_epoch](https://github.com/microsoft/hi-ml/blob/ce1535b5958c10a1c883411578247c934befce9e/hi-ml-histopathology/src/histopathology/models/deepmil.py#L297)
does it make sense? Happy to chat more if you have a simpler suggestion

Reply:
Yes makes sense. Would it be simpler to save these outputs in `on_validation_end` rather than `on_validation_epoch_end`?